File: src/main/java/wpd2.groupm/Models/Project.java
package wpd2.groupm.Models;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Random;

public class Project {
    private int id;
    private String name, description;
    private Date ExpectedDone, actualDone;
    private List<Milestone> milestones;

    public Project() {
        Random rand = new Random();
        this.id = rand.nextInt(100000);
    }
    public Project(String name,String description) {
        this.name = name;
        this.description = description;
        this.milestones = new ArrayList<>();
    }
    public Project(String name,String description,int id) {
        this.name = name;
        this.description = description;
        this.id = id;
        this.milestones = new ArrayList<>();
    }
    public Project(String name,String description,int id,List<Milestone> milestones) {
        this.name = name;
        this.description = description;
        this.id = id;
        this.milestones = milestones;
    }

    public List<Milestone> getmilestones() {
        return milestones;
    }

    public void setmilestones(List<Milestone> milestones) {
        this.milestones = milestones;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getExpectedDone() {
        return ExpectedDone;
    }

    public void setExpectedDone(Date expectedDone) {
        ExpectedDone = expectedDone;
    }

    public Date getActualDone() {
        return actualDone;
    }

    public void setActualDone(Date actualDone) {
        this.actualDone = actualDone;
    }
}

File: model/project.js
class project{
  let id;
  let userId;
  let name;
  let description;
  let date expectedDone = new Date();
  let date actualDone = new Date();
  let milestoneList = [];

  constructor(){
    //set the id of the project
    //not sure how to do that right now
  }

  constructor(name, description){
    this.name = name;
    this.description = description;
  }

  constructor(name, description){
    this.name = name;
    this.description = description;
    this.milestoneList = //new array
  }

  constructor(name, description, id, , userId, milestoneList)
    this.name = name;
    this.description = description;
    this.id = id;
    this.milestoneList = milestoneList;
    this.userId;
  }

  get milestoneList(milestoneList){
    return this.milestoneList;
  }

  set milestoneList(milestoneList){
    this.milestoneList = milestoneList;
  }

  get id(id){
    return this.id;
  }

  set id(id){
    this.id = id;
  }

  get name(name){
    return this.name;
  }

  set name(name){
    this.name = name;
  }

  get description(description){
    return  this.description;
  }

  set description(description){
    this.description = description;
  }

  get expectedDone(expectedDone){
    return this.expectedDone;
  }

  set expectedDone(expectedDone){
    this.expectedDone = expectedDone;
  }

  get actualDone(actualDone){
    return this.actualDone;
  }

  set actualDone(actualDone){
    this.actualDone = actualDone;
  }

  get userId(userId){
    return this.userId;
  }

  set userId(userId){
    this.userId = userId;
  }
